sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -532,6 +532,20 @@ object GpuOverrides {
         // There are so many of these that we don't need to print them out.
         override def print(append: StringBuilder, depth: Int, all: Boolean): Unit = {}
       }),
+    // TODO: maybe we should apply the conversion between decimal32 and decimal64 to promote precision?
+    expr[PromotePrecision](
+      "Eliminates unnecessary PromotePrecision expressions in GPU runtime",
+      (a, conf, p, r) => new UnaryExprMeta[PromotePrecision](a, conf, p, r) {
+        // Remove PromotePrecision and (GPU)Cast expression appended by DecimalPrecision
+        override def convertToGpu(child: Expression): GpuExpression = {
+          child match {
+            case c: GpuCast => c.child.asInstanceOf[GpuExpression]
+            case c => throw new IllegalStateException(
```

Comment:
Is this guaranteed 100% of the time that the child will be a cast?  If not then we need to figure out better way to handle this?

Reply:
So far, it is 100% guaranteed since case class `PrecisionPromotion`will only be created inside private method `DecimalPrecision.promotePrecision`with `Cast` Op.